### ISSUE TITLE
Support wkhtmltopdf 0.12.6

### ISSUE THREAD

**Issue by harri35** [MobSF/Mobile-Security-Framework-MobSF#1478]
ENVIRONMENT

```
OS and Version: Windows 10 version 1909
Python Version: Python 3.6.2
MobSF Version: v3.0.9 Beta
```

## EXPLANATION OF THE ISSUE
When I try to generate a PDF report of a static analysis I get the following error from the MobSF UI

```
{"pdf_error": "Cannot Generate PDF", "err_details": "wkhtmltopdf reported an error:\nExit with code 1 due to network error: ProtocolUnknownError\r\n"}
```
I just installed wkhtmltopdf. Also added it to path. Also tried to reinstall it.
Just calling  
```
wkhtmltopdf .\test.html test.pdf
```
were the test.html is
```
<!DOCTYPE html>
<html>
<body>

<h1>My First Heading</h1>

<p>My first paragraph.</p>

</body>
</html>
```
works fine and generates a pdf.

## STEPS TO REPRODUCE THE ISSUE

```
1. Analyse an apk
2. Generate a pdf
3. An error is returned
```

## LOG FILE

```
[INFO] 21/Jul/2020 10:58:26 - 
  __  __       _    ____  _____       _____  ___  
 |  \/  | ___ | |__/ ___||  ___|_   _|___ / / _ \ 
 | |\/| |/ _ \| '_ \___ \| |_  \ \ / / |_ \| | | |
 | |  | | (_) | |_) |__) |  _|  \ V / ___) | |_| |
 |_|  |_|\___/|_.__/____/|_|     \_/ |____(_)___/

[INFO] 21/Jul/2020 10:58:26 - Mobile Security Framework v3.0.9 Beta
[INFO] 21/Jul/2020 10:58:26 - OS: Windows
[INFO] 21/Jul/2020 10:58:26 - Platform: Windows-10-10.0.18362-SP0
[INFO] 21/Jul/2020 10:58:26 - Dist:   
[INFO] 21/Jul/2020 10:58:26 - MobSF Basic Environment Check
[INFO] 21/Jul/2020 10:58:27 - Checking for Update.
[INFO] 21/Jul/2020 10:58:28 - No updates available.
[INFO] 21/Jul/2020 10:58:33 - Fetching data from DB for PDF Report Generation (Android)
[INFO] 21/Jul/2020 10:58:33 - Analysis is already Done. Fetching data from the DB...
[INFO] 21/Jul/2020 10:58:33 - Generating PDF report for android apk
[ERROR] 21/Jul/2020 10:58:35 - Error Generating PDF Report
Traceback (most recent call last):
  File "D:\m\gitrepo\Mobile-Security-Framework-MobSF\StaticAnalyzer\views\shared_func.py", line 190, in pdf
    pdf_dat = pdfkit.from_string(html, False, options=options)
  File "d:\m\gitrepo\mobile-security-framework-mobsf\venv\lib\site-packages\pdfkit\api.py", line 72, in from_string
    return r.to_pdf(output_path)
  File "d:\m\gitrepo\mobile-security-framework-mobsf\venv\lib\site-packages\pdfkit\pdfkit.py", line 156, in to_pdf
    raise IOError('wkhtmltopdf reported an error:\n' + stderr)
OSError: wkhtmltopdf reported an error:
Exit with code 1 due to network error: ProtocolUnknownError


[ERROR] 21/Jul/2020 10:58:35 - Internal Server Error: /PDF/
```

**Comment by harri35** [comment]
I installed MobSF to a VirtualBox VM with Ubuntu 19.10 eoan and pdf generation works fine.
I guess this is either my Windows setup specific issue or a Windows-specific issue

**Comment by ajinabraham** [comment]
Is `wkhtmltopdf http://google.com google.pdf` working for you in windows?

**Comment by ArivuLabs** [comment]
Hi All,

I am getting the same error:

OS and Version: **Mac OS:  10.15.3 (19D76)**
Python Version: **Python 3.7**
MobSF Version: **v3.0.9 Beta**

**{"pdf_error": "Cannot Generate PDF", "err_details": "wkhtmltopdf reported an error:\nExit with code 1 due to network error: ProtocolUnknownError\n"}**

**Comment by ArivuLabs** [comment]
> Is `wkhtmltopdf http://google.com google.pdf` working for you in windows?

**I am able to run on my terminal:**  wkhtmltopdf http://google.com google.pdf

**Comment by ajinabraham** [comment]
Works fine at my end. What version of wkhtmltopdf are you using? Is it Qt patched?
`Exit with code 1 due to network error: ProtocolUnknownError`  shows up when you have network connectivity issues or when you are behind a proxy.

Here is mine.
```
$ wkhtmltopdf --version
wkhtmltopdf 0.12.5 (with patched qt)
```

**Comment by ArivuLabs** [comment]
> Works fine at my end. What version of wkhtmltopdf are you using? Is it Qt patched?
> `Exit with code 1 due to network error: ProtocolUnknownError` shows up when you have network connectivity issues or when you are behind a proxy.
> 
> Here is mine.
> 
> ```
> $ wkhtmltopdf --version
> wkhtmltopdf 0.12.5 (with patched qt)
> ```

**My Version:** 
wkhtmltopdf 0.12.6 (with patched qt)

**Comment by ajinabraham** [comment]
I have seen more people reporting issues w.r.t to pdf generation. Do you mind installing 0.12.5 and see if the issue still exist? I just want to make sure if it is a wkhtmltopdf issue or something at your end.

**Comment by ArivuLabs** [comment]
> I have seen more people reporting issues w.r.t to pdf generation. Do you mind installing 0.12.5 and see if the issue still exist? I just want to make sure if it is a wkhtmltopdf issue or something at your end.

**Hi @ajinabraham  with I am able to run the report with "wkhtmltopdf 0.12.5 (with patched qt)"** :)

Thanks

**Comment by ajinabraham** [comment]
Ok so looks like something changed in 0.12.6.

**Comment by ajinabraham** [comment]
Fix https://github.com/wkhtmltopdf/wkhtmltopdf/issues/4536#issuecomment-643019765

**Comment by ajinabraham** [comment]
This will be addressed in `3.1.0`
